irohad/consensus/yac/impl/yac_gate_impl.cpp

```diff
@@ -64,7 +67,8 @@ namespace iroha {
                      current_hash_.vote_hashes.block_hash);
         }
 
-        auto order = orderer_->getOrdering(current_hash_);
+        auto order = orderer_->getOrdering(current_hash_,
+                                           *event.ledger_state->ledger_peers);
```

Comment:
Can't the shuffling be done in a more optimal way, for example, via passing not a copy of `std::vector<std::shared_ptr>>`, but `std::vector<int>`, where int-s are indices of peers to be shuffled? Then this order can be passed to `vote`, then those indices can be matched with an original collection, when `vote`-ing

Reply:
Probably this can be done after profiling if it turns out to be a bottleneck.